docs/proof_of_life_contact_list_app.md

```diff
@@ -0,0 +1,469 @@
+
+**Evolution Plan**: As we evolve past our proof-of-life iteration, we will evolve this document to flesh out the necessary features for our next steps. The purpose of this document is to:
+
+- Cover the work we're doing right now
+- Evolve to reflect the details of the work to come
+- Maintain a living reference for implementation teams
+
+**Alignment Note**: This implementation plan directly implements the principles and architecture outlined in [`5_mobile_native_app_generation.md`](./5_mobile_native_app_generation.md), serving as a concrete proof-of-concept for the instruction-first, three-phase workflow approach.
+
+## Success Criteria
+
+**User Journey**: From utterance _"I want an iOS mobile app that will show me a list of all of my Salesforce Contacts"_ to a running mobile app displaying Contacts in a simulator/emulator.
+
+**Technical Validation**: Demonstrates instruction-first MCP tools successfully guiding an LLM through the complete Plan → Design/Iterate → Run workflow.
+
+---
+
+## Prerequisites and Assumptions
+
+### Environment Assumptions (Happy Path Focus)
+
+- User is on a supported development platform (macOS for iOS development, macOS/Windows/Linux for Android development)
+- User is using Cursor IDE for agentic development
+- Pre-installed tools:
+  - `@salesforce/mobile-native-mcp-server` MCP server
+  - Salesforce CLI with `sfdx-mobilesdk-plugin`, `@salesforce/lwc-dev-mobile`, and `@salesforce/lwc-dev-mobile-core` plugins
+  - Platform-specific development tools will be validated based on target platform (Xcode for iOS, Android Studio for Android)
+- User has a pre-configured Salesforce Connected App with known Client ID and Callback URI
+
+### Resource Paths
+
+- `<ServerRoot>`: Filesystem path to `@salesforce/mobile-native-mcp-server` package contents
+- Templates located at: `<ServerRoot>/resources/SalesforceMobileSDK-Templates/` (Git submodule pointing to [SalesforceMobileSDK-Templates](https://github.com/forcedotcom/SalesforceMobileSDK-Templates))
+
+---
+
+## High-Level Agentic Workflow
+
+### Instruction-First Approach
+
+1. **User utterance**: _"I want an iOS mobile app that will show me a list of all of my Salesforce Contacts."_
+
+2. **LLM calls Workflow Orchestration Guide** → receives complete roadmap for Contact list app workflow
+
+3. **Phase 1: Plan** (Template Discovery → Project Generation → Build Validation)
+   - LLM guided through template discovery and selection
+   - LLM guided through project creation with Connected App configuration
+   - LLM guided through build verification
+
+4. **Phase 2: Design/Iterate** (Code Analysis → Feature Modification)
+   - LLM guided through exploring template structure
+   - LLM guided through adapting record list for Contacts
+
+5. **Phase 3: Run** (Deployment)
+   - LLM guided through deploying to iOS simulator
+
+---
+
+## MCP Server Tool Specifications
+
+### Primary Orchestration Tool
+
+#### `sfmobile-native-workflow-guide`
```

Comment:
how do we ensure this is the first tool that gets executed? should each tools description indicate this as a pre-req?

Reply:
I added [a section](https://github.com/forcedotcom/mobile-mcp-tools/pull/46/files#diff-6618e0283c299b0004f135955408f87664c70b937d116e29e2d1fe5090615b60R468-R497) below to capture this requirement.